src/main/java/swift/ui/MainWindow.java

```diff
@@ -180,40 +181,45 @@ private void handleExit() {
      */
     @FXML
     private void handleTaskTab() {
-        logic.showAllLists();
-
-        personListPanel.removeHeading();
-        taskListPanel = new TaskListPanel(logic.getFilteredTaskList(), logic.getFilteredBridgeList(),
+        taskListPanel = new TaskListPanel(logic.getFilteredTaskList(), logic.getUnfilteredBridgeList(),
                 logic.getFilteredPersonList());
+
+        listPanelPlaceholder.getChildren().clear();
         listPanelPlaceholder.getChildren().add(taskListPanel.getRoot());
 
-        personTaskListPanel.removeHeading();
-        taskPersonListPanel = new TaskPersonListPanel(logic.getFilteredPersonList());
-        personTaskListPanelPlaceholder.getChildren().add(taskPersonListPanel.getRoot());
+        if (logic.getFilteredTaskList().size() == 1) {
```

Comment:
Can add a comment above to explain what you are doing over here because I know what it does but not really explicit in what it does